### ISSUE TITLE
Order of destructuring issue

### ISSUE THREAD

**Issue by bmeck** [6to5/6to5#549]
Order of destructuring / defaulting of arguments fails for dependent arguments. 

```
function required(msg) {
  throw new Error(msg);
}
function sum({
    arr = required('arr is required')
  }
  = {
    arr: arr = []
  },
  length = arr.length
) {
  let i = 0;
  let acc = 0;
  for (let item of arr) {
    if (i >= length) return acc;
    acc += item;
    i++;
  }
  return acc;
}

console.log(sum({arr:[1,2]}))
```

Expected `3`
Got `Cannot read property 'length' of undefined`

**Comment by sebmck** [comment]
Fixed as of 2.13.4, thanks!

**Comment by bmeck** [comment]
<3
